lndclient/lnd_services.go

```diff
@@ -40,13 +38,25 @@ type GrpcLndServices struct {
 }
 
 // NewLndServices creates a set of required RPC services.
-func NewLndServices(lndAddress string, application string,
-	network string, macPath, tlsPath string) (
-	*GrpcLndServices, error) {
+func NewLndServices(lndAddress, application, network, macaroonDir,
+	tlsPath string) (*GrpcLndServices, error) {
+
+	// If the macaroon directory isn't set, then we can't proceed as we
+	// need then to obtain the macaroons for all sub-servers.
+	if macaroonDir == "" {
```

Comment:
Do we still want to allow running without macaroons? This was possible before.

Reply:
IMO no, as we don't really recommend running `lnd` itself without macaroons. We may want to add a way to specify the raw hex macaroons over the command line though. Adding this line helped me find some bugs in the prior implementation, and for example will make things fail if users run on `regtest`.

Reply:
Actually ended up reverting this in order to consolidate the default macaroon directory code for re-use.